### PR TITLE
More error handling.

### PR DESCRIPTION
**Pull request by glipR**
Fixes problems with students malforming their workspace path (About 70% of sentry crashes)
Makes certain problems with bots do more than just error out, shows an error window.
Overall a bit safer than previous versions.

### CODE REVIEW

**Comment by glipR** [comment]
@jtbui20 @angustrau @NervousDendrite 
Should fix most issues we ran into today (Not sure about the downloading custom tasks one, but maybe?).
Also makes it a lot easier for me to make further "bug"-fixes (Fixing students renaming folders, deleting files and whatnot).

If someone could just have a read through my code and make sure everything looks ok happy to merge and release.

File: ev3sim/gui.py [hunk 1]
```diff
@@ -89,6 +95,20 @@ def main(passed_args=None):
...
         args.__dict__.update(passed_args)
 
+    if args.open_config:
+        import platform
+        import subprocess
+
+        fname = os.path.join(os.path.dirname(os.path.abspath(__file__)), "user_config.yaml")
+
+        if platform.system() == "Windows":
+            subprocess.Popen(["explorer", "/select,", fname])
```
Comment (angustrau) [inline ev3sim/gui.py:105]: Is this a typo?

Comment (glipR) [reply ev3sim/gui.py:105]: Doesn't seem so https://stackoverflow.com/questions/281888/open-explorer-on-a-file

**Comment by glipR** [comment]
@angustrau all good with this? want to get it merged before thursday.

**Review by angustrau — APPROVED** [review]

**Pull request merged**